src/components/POLine/hooks/useIsFundsRestrictedByLocationIds/useIsFundsRestrictedByLocationIds.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-shadow */
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { useFundsById } from '../useFundsById';
+import { useLocationsByHoldingIds } from '../useLocationsByHoldingIds';
+
+export const useIsFundsRestrictedByLocationIds = ({
+  fundIds = [],
+  locationIds = [],
+  holdingIds = [],
+}) => {
+  const [hasLocationRestrictedFund, setHasLocationRestrictedFund] = useState(false);
+
+  const {
+    isLoading: isLocationLoading,
+    locationIds: permanentLocationId,
+  } = useLocationsByHoldingIds(holdingIds);
+
+  const listOfLocationIDs = useMemo(() => [...locationIds, ...permanentLocationId], [locationIds, permanentLocationId]);
+
+  const { funds, isLoading: isFundsLoading } = useFundsById(fundIds, {
+    enabled: !isLocationLoading,
+  });
+
+  const fundWithRestrictedLocations = useMemo(() => {
+    return funds
+      .filter(({ restrictByLocations }) => restrictByLocations)
+      .map(({ locationIds, id }) => ({ id, locationIds }));
+  }, [funds]);
+
+  const isFundNotRestricted = useCallback(() => {
+    return listOfLocationIDs.some((locationId) => {
+      return fundWithRestrictedLocations.some(({ locationIds }) => locationIds.includes(locationId));
+    });
+  }, [fundWithRestrictedLocations, listOfLocationIDs]);
+
+  useEffect(() => {
+    if (fundWithRestrictedLocations.length) {
+      const fundRestricted = !isFundNotRestricted();
+
+      setHasLocationRestrictedFund(fundRestricted);
+    }
+  }, [fundWithRestrictedLocations, hasLocationRestrictedFund, isFundNotRestricted]);
```

Comment:
Looks great! However, it is possible not to use `useEffect`, the result is a boolean anyway. 

If `useEffect` is utilized to reduce calculation in `isFundNotRestricted`, then `useMemo` can be applied. The side effect seems excessive.

Reply:
done